From: Andrew Morton <akpm@linux-foundation.org>
Subject: minmax-add-in_range-macro-fix
Date: Tue Jul 11 01:02:31 PM PDT 2023

fix lib/logic_pio.c

lib/logic_pio.c:24: warning: "in_range" redefined
   24 | #define in_range(b, first, len)        ((b) >= (first) && (b) < (first) + (len))
      | 
In file included from ./include/linux/kernel.h:27,
                 from ./include/linux/cpumask.h:10,
                 from ./include/linux/smp.h:13,
                 from ./include/linux/lockdep.h:14,
                 from ./include/linux/mutex.h:17,
                 from ./include/linux/kernfs.h:11,
                 from ./include/linux/sysfs.h:16,
                 from ./include/linux/kobject.h:20,
                 from ./include/linux/of.h:17,
                 from lib/logic_pio.c:11:
./include/linux/minmax.h:183: note: this is the location of the previous definition
  183 | #define in_range(val, start, len)                                       \
      | 

Cc: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/logic_pio.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/lib/logic_pio.c~minmax-add-in_range-macro-fix
+++ a/lib/logic_pio.c
@@ -12,6 +12,7 @@
 #include <linux/io.h>
 #include <linux/logic_pio.h>
 #include <linux/mm.h>
+#include <linux/minmax.h>
 #include <linux/rculist.h>
 #include <linux/sizes.h>
 #include <linux/slab.h>
@@ -20,9 +21,6 @@
 static LIST_HEAD(io_range_list);
 static DEFINE_MUTEX(io_range_mutex);
 
-/* Consider a kernel general helper for this */
-#define in_range(b, first, len)        ((b) >= (first) && (b) < (first) + (len))
-
 /**
  * logic_pio_register_range - register logical PIO range for a host
  * @new_range: pointer to the IO range to be registered.
_
